src/compiler/transformers/declarations.ts

```diff
@@ -1947,6 +1948,32 @@ export function transformDeclarations(context: TransformationContext) {
         // Anything left unhandled is an error, so this should be unreachable
         return Debug.assertNever(input, `Unhandled top-level node in declaration emit: ${Debug.formatSyntaxKind((input as Node).kind)}`);
 
+        function isExprEnumMember(enumDecl: EnumDeclaration, expr: Expression): boolean {

```

Comment:
We do support expressions in enum members so `enum E { A = 1<< 10 }` would be valid. I think you should try adapt the evaluator in [checker](https://github.dev/bloomberg/TypeScript/blob/1e141504c7c6f38927b2581b9ed3f75aec3a0213/src/compiler/checker.ts#L44691). Ideally merge it with the copied evaluator in the [DTE](https://github.dev/bloomberg/TypeScript/blob/1e141504c7c6f38927b2581b9ed3f75aec3a0213/src/compiler/transformers/declarations/evaluator.ts#L20-L21)